### PR TITLE
Mix training and inference infra and manifests

### PR DESCRIPTION
**Pull request by TarasRudko**
## Description

This PR....

## Tasks

* [ ] The [contributing guide](https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/blob/main/.github/CONTRIBUTING.md) has been read and followed.
* [ ] The samples added / modified have been fully tested.
* [ ] Workflow files have been added / modified, if applicable.
* [ ] Region tags have been properly added, if new samples.
* [ ] All dependencies are set to up-to-date versions, as applicable.
* [ ] Merge this pull-request for me once it is approved.

### CODE REVIEW

**Review by kenthua — COMMENTED** [review]
can we remove the `src` folder?

**Comment by TarasRudko** [comment]
> can we remove the `src` folder?

Done

File: ai-ml/mix-train-and-inference/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    --member=principal://iam.googleapis.com/projects/$PROJECT_NUMBER/locations/global/workloadIdentityPools/$PROJECT_ID.svc.id.goog/subject/ns/$NAMESPACE/sa/default \
+    --condition=None
+gcloud storage buckets add-iam-policy-binding "gs://$TRAINING_DATA_BUCKET" \
+    --role=roles/storage.objectViewer \
+    --member=principal://iam.googleapis.com/projects/$PROJECT_NUMBER/locations/global/workloadIdentityPools/$PROJECT_ID.svc.id.goog/subject/ns/$NAMESPACE/sa/default \
+    --condition=None
+
+
+gcloud artifacts repositories add-iam-policy-binding fine-tuning \
+    --role=roles/artifactregistry.reader \
```
[the remark this answers is not in the document]
Comment (TarasRudko) [reply ai-ml/mix-train-and-inference/deploy.sh]: yes, not needed anymore. Corrected

File: ai-ml/mix-train-and-inference/kueue/patch.yaml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    #  requeuingStrategy:
+    #    timestamp: Eviction
+    #    backoffLimitCount: null # null indicates infinite requeuing
+    #    backoffBaseSeconds: 60
+    #    backoffMaxSeconds: 3600
+    #manageJobsWithoutQueueName: true
+    #internalCertManagement:
+    #  enable: false
+    #  webhookServiceName: ""
+    #  webhookSecretName: ""
```
Comment (NimJay) [inline ai-ml/mix-train-and-inference/kueue/patch.yaml:58]: **Question:** There are some lines commented out here. Is this intentional?

Comment (kenthua) [reply ai-ml/mix-train-and-inference/kueue/patch.yaml:58]: Yes, the team felt it was important to keep the context of the comments.

File: ai-ml/mix-train-and-inference/workloads/fine-tune-l4.yaml [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        app: finetune-job
+      annotations:
+        gke-gcsfuse/volumes: "true"
+        gke-gcsfuse/memory-limit: "35Gi"
+    spec:
+      priorityClassName: low-priority-preempting
+      containers:
+      - name: gpu-job
+        imagePullPolicy: Always 
+        image: us-docker.pkg.dev/google-samples/containers/gke/gemma-fine-tuning:v1.0.0
```
Comment (NimJay) [inline ai-ml/mix-train-and-inference/workloads/fine-tune-l4.yaml:48]: Just a note for future developers (any passerby): This source code for this image exists at https://github.com/GoogleCloudPlatform/accelerated-platforms/tree/974f2eff748d00d2566024d6ec4dd7f309f641c5/use-cases/model-fine-tuning-pipeline/fine-tuning/pytorch/src.

File: ai-ml/mix-train-and-inference/remove.sh [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+cd gke-platform
+sed -ie 's/"deletion_protection": true/"deletion_protection": false/g' terraform.tfstate
```
Comment (NimJay) [inline ai-ml/mix-train-and-inference/remove.sh:18]: Nitpick (feel free to ignore): Consider setting `deletion_protection` to `false` from the very start (in the [google_container_cluster resources](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster#deletion_protection-1)).

File: ai-ml/mix-train-and-inference/gke-platform/modules/gke_standard/main.tf
Comment (NimJay) [inline ai-ml/mix-train-and-inference/gke-platform/modules/gke_standard/main.tf:1]: Just a thought. This is out-of-scope for this pull-request, but we should consider modularizing the Terraform in this git repo. These gke_standard and gke_autopilot folders look similar to [existing gke_standard](https://github.com/search?q=repo%3AGoogleCloudPlatform%2Fkubernetes-engine-samples%20gke_standard&type=code) and [existing gke_autopilot](https://github.com/search?q=repo%3AGoogleCloudPlatform%2Fkubernetes-engine-samples%gke_autopilot&type=code). I'll add a comment in https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/issues/861.

**Review by NimJay — APPROVED** [review]
Awesome work on these samples! 👏 
Looks good to me.
I trust you've tested these samples appropriately for functionality.

Left a few comments, but nothing major.
Judging from internal discussions, I'm guessing this is ready for merge (even though it's still in draft mode). Merging...

**Pull request merged**